File: 백준/Silver I/11057. 오르막 수/오르막 수.cc
#include <iostream>
using namespace std;

int main()
{
    int N;
    cin >> N;
    
    int dp[10] = {0};
    dp[0] = 1;

    for (int i = 0; i < N; i++)
    {
        for (int j = 1; j < 10; j++)
        {
            dp[j] += dp[j-1];
            dp[j] %= 10007;
        }
    }

    int answer = 0;
    
    for (int k = 0; k < 10; k++)
    {
        answer += dp[k];     
        answer %= 10007;
    }

    cout << answer;
    return 0;
}

File: 백준/Silver I/11057. 오르막 수/오르막 수.cs
using System;
namespace MySolution
{
    class Solution
    {
        public static void Main()
        {
            int N = int.Parse(Console.ReadLine());
            int[] dp = new int[10];
            dp[0] = 1;

            for (int i = 0; i < N; i++)
            {
                for (int j = 1; j < 10; j++)
                {
                    dp[j] += dp[j-1];
                    dp[j] %= 10007;
                }
            }

            int answer = 0;
            for (int i = 0; i < 10; i++)
            {
                answer += dp[i];
                answer %= 10007;
            }

            Console.WriteLine(answer);
        }
    }
}